Userland/Libraries/LibGfx/ImageFormats/ExifReader.h

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2023, Lucas Chollet <lucas.chollet@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Endian.h>
+#include <AK/MemoryStream.h>
+#include <AK/String.h>
+
+// This implements "Exchangeable image file format for digital still cameras: Exif Version 3.0"
+// The spec can be found at https://www.cipa.jp/e/std/std-sec.html
+
+namespace Gfx {
+
+struct ExifMetadata {
+};
+
+class ExifReader {
+public:
+    static ErrorOr<ExifMetadata> read(FixedMemoryStream& stream);
+
+private:
+    enum class ByteOrder {
+        LittleEndian,
+        BigEndian,
+    };
+
+    enum class Type {
+        Byte = 1,
+        ASCII = 2,
+        Short = 3,
+        Long = 4,
+        Rational = 5,
+        Undefined = 7,
+        SLong = 9,
+        SRational = 10,
+        UTF8 = 129,
+    };
+
+    template<OneOf<u32, i32> x32>
```

Comment:
Having to specify the template argument every time, when there are only signed/unsigned Rationals, makes me wonder if a couple of type aliases (and calling this something like RationalImpl) would be nicer.

Reply:
Or just make two structs, seems easier.